packages/devtools_app/lib/src/flutter/scaffold.dart

```diff
@@ -172,19 +171,25 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
     }
   }
 
-  /// Pushes a screen for an offline import.
-  void _pushScreenForImport(DevToolsScreenType screenType) {
+  /// Pushes the snapshot screen for an offline import.
+  void _pushSnapshotScreenForImport(
+    String screenId,
+    Object data,
+  ) {
+    final args = SnapshotArguments(screenId, data);
+    if (offlineMode) {
+      // If we are already in offline mode, only handle routing from existing
```

Comment:
what happens if we don't do this?

Reply:
Added a comment to explain. copying and pasting here:

If we allow other routes that are not the '/snapshot' route to handle
       routing when we are already offline, the other routes will pop their
       existing screen ('/connect', or '/') and push '/snapshot' over the top.
       We want to avoid this because the routes underneath the existing
       '/snapshot' route should remain unchanged while '/snapshot' sits on
       top.